File: AndroidMVVMJsonExample/app/src/main/java/com/blackbirds/shakil/androidmvvmjsonexample/MainActivity.java
package com.blackbirds.shakil.androidmvvmjsonexample;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;

import com.blackbirds.shakil.androidmvvmjsonexample.Adapter.MovieListAdapter;
import com.blackbirds.shakil.androidmvvmjsonexample.ViewModel.MainActivityViewModel;

public class MainActivity extends AppCompatActivity {

    MainActivityViewModel mainActivityViewModel;

    RecyclerView recyclerView;
    MovieListAdapter adapter;
    LinearLayoutManager layoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainActivityViewModel = new ViewModelProvider(this).get(MainActivityViewModel.class);

        recyclerView = findViewById(R.id.recyclerView);
        layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);

        mainActivityViewModel.getMovieList().observe(this, movieModels -> {
            Log.e("MainActivity", "MovieList: " + movieModels.get(0).getName());

            if (movieModels != null){
                adapter = new MovieListAdapter(this, movieModels);
                adapter.notifyDataSetChanged();
                recyclerView.setAdapter(adapter);
            }
        });
    }
}

File: KotlinMVVMJsonExample/app/src/main/java/com/blackbirds/shakil/kotlinmvvmjsonexample/MainActivity.kt
package com.blackbirds.shakil.kotlinmvvmjsonexample

import android.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.blackbirds.shakil.kotlinmvvmjsonexample.Adapter.MovieListAdapter
import com.blackbirds.shakil.kotlinmvvmjsonexample.ViewModel.MainActivityViewModel
import dmax.dialog.SpotsDialog

class MainActivity : AppCompatActivity() {

    var mainActivityViewModel: MainActivityViewModel? = null
    var recyclerView: RecyclerView? = null
    var adapter: MovieListAdapter? = null
    var layoutManager: LinearLayoutManager? = null
    var dialog: AlertDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mainActivityViewModel = ViewModelProvider(this)[MainActivityViewModel::class.java]

        dialog = SpotsDialog.Builder().setCancelable(false).setContext(this).build()
        dialog!!.show()

        recyclerView = findViewById(R.id.recyclerView)
        recyclerView!!.setHasFixedSize(true)
        layoutManager = LinearLayoutManager(this)
        recyclerView!!.layoutManager = layoutManager

        mainActivityViewModel!!.getMovieList.observe(this) { movieModels ->
            Log.e("MainActivity", "MovieList: " + movieModels.get(0).name)

            if (movieModels != null){
                adapter = MovieListAdapter(this, movieModels)
                adapter!!.notifyDataSetChanged()
                recyclerView!!.adapter = adapter
                dialog!!.dismiss()
            }
        }
    }
}